osaca/parser/parser_AArch64.py

```diff
@@ -326,6 +350,8 @@ def process_memory_address(self, memory_address):
         """Post-process memory address operand"""
         # Remove unnecessarily created dictionary entries during parsing
         offset = None if 'offset' not in memory_address else memory_address['offset']
+        if isinstance(offset, list) and len(offset) == 1:
+            offset = offset[0]
```

Comment:
does it mean that `memory.offset` may be a list?

Reply:
In some cases, pyparsing is returning offset as a list (I think this is due to some combination of `pp.Optional()` with different options).
For example for: `ldr x0, [x0, #:got_lo12:q2c]`.
The post-processing methods for each operand type are used for this kind of "clean up" in many ways

Reply:
If the list is always one element long, that's fine because it will always be a single value. Otherwise I would suggest always returning a list.

Reply:
offset itself can be a dictionary with multiple elements, but there should be never one operand with more than one offset.